Content.Server/Disposal/Unit/EntitySystems/DisposalUnitSystem.cs

```diff
@@ -187,11 +182,12 @@ public void DoInsertDisposalUnit(EntityUid unit, EntityUid toInsert, DisposalUni
         public override void Update(float frameTime)
         {
             base.Update(frameTime);
-            for (var i = _activeDisposals.Count - 1; i >= 0; i--)
+            foreach (var (_, comp) in EntityQuery<ActiveDisposalUnitComponent, DisposalUnitComponent>())
             {
-                var comp = _activeDisposals[i];
-                if (!Update(comp, frameTime)) continue;
-                _activeDisposals.RemoveAt(i);
+                if (!Update(comp, frameTime))
+                    continue;
+
+                RemComp<ActiveDisposalUnitComponent>(comp.Owner);
```

Comment:
```suggestion
                RemCompDeferred<ActiveDisposalUnitComponent>(comp.Owner);
```
this will modifying enumerated collection moment otherwise im pretty sure

Reply:
It's fine to do. I forget when exactly it changed (I also tested it ingame).

Reply:
if you say so